web-console/src/lib/services/pipelineManagerQuery.ts

```diff
@@ -46,7 +46,26 @@ const toClientPipelineStatus = (status: RawPipelineStatus) => {
     .exhaustive()
 }
 
-export const PipelineManagerQuery = (({ pipelines, ...queries }) => ({
+/**
+ * Consolidate local STARTING | PAUSING | SHUTTING_DOWN status with remote PROVISIONING status
+ * @param newPipeline
+ * @param oldPipeline
+ * @returns
+ */
+const consolidatePipelineStatus = (newPipeline: Pipeline, oldPipeline: Pipeline | undefined) => {
+  const current_status =
```

Comment:
can we make this more readable? e.g. something like 
```suggestion
  const isNotDesired = newPipeline.state.current_status !== toClientPipelineStatus(newPipeline.state.desired_status)
  const isNotFailed = newPipeline.state.current_status !== PipelineStatus.FAILED
  const oldHasStatus = oldPipeline?.state.current_status
  const oldWasTransitioning = oldHasStatus && [PipelineStatus.STARTING, PipelineStatus.PAUSING, PipelineStatus.SHUTTING_DOWN].includes(oldPipeline.state.current_status)
  
  const current_status =
    isNotDesired &&
    isNotFailed &&
    oldWasTransitioning
      ? oldPipeline.state.current_status
      : newPipeline.state.current_status
  return updatePipelineStatus('current_status', current_status)(newPipeline)
```

Reply:
Let me see

Reply:
Nice idea, added